packages/opam-update/opam-update.1.2.1/opam

```diff
@@ -0,0 +1,34 @@
+opam-version: "1.2"
+maintainer: "louis.gesbert@ocamlpro.com"
+authors: [
+  "Thomas Gazagnaire <thomas@gazagnaire.org>"
+  "Louis Gesbert <louis.gesbert@ocamlpro.com>"
+]
+homepage: "https://opam.ocaml.org"
+bug-reports: "https://github.com/ocaml/opam/issues"
+dev-repo: "https://github.com/ocaml/opam.git"
+build: [
+  ["./configure"]
+  ["make" "lib-ext"]
+  [make "opam"]
+]
+install: [
+  ["sh" "-c" "echo \"%{version}%\" > \"%{root}%/opam.version\""]
+  ["install" "-m" "755" "src/opam" "%{root}%/opam"]
+]
+# Restrict self-upgrade to "official" versions, just in case
+available: [
+  opam-version >= "1.2" & opam-version <= version &
+  (compiler = "3.12.1" |
+   compiler = "4.00.0" |
+   compiler = "4.00.1" |
+   compiler = "4.01.0" |
+   compiler = "4.02.0" |
+   compiler = "4.02.1" |
+   (compiler = "system" & ocaml-version >= "3.12.1" & ocaml-version < "4.03"))
+]
+post-messages: [
+"IMPORTANT: OPAM %{version}% has been installed to %{root}%, and it will be used by default for ALL switches. To avoid accidentally downgrading, removing the package WON'T revert this.
```

Comment:
We could also make the uninstallation *really* smooth by installing an `opam-upgrade` plugin that performs the correct removal, rather than listing the easily forgotten post-messages.  In particular, I'm not sure what `opam.version` does and what the consequences are of removing just the `opam` binary and not the associated version file, or what happens if they were to get out of sync.